specs/lib/removeUnusedVerticesSpec.js

```diff
@@ -419,6 +419,200 @@ describe('removeUnusedVertices', function() {
             oldTextureBuffer.slice(96, 160)], 96))).toBe(true);
     });
 
+    it('removes parts of the buffer based on the attribute type if the stride is 0', function(){
+        var indices = [0,1,2,0,2,3];
+        var indicesBuffer = new Buffer(indices.length * 2);
+        for (var i = 0; i < indices.length; i++) {
+            indicesBuffer.writeUInt16LE(indices[i], i * 2);
+        }
+
+        var positions = [
+            0,0,0,
+            0,1,0,
+            1,1,0,
+            1,0,0,
+            2,2,2,
+            2,2,2,
+            2,2,2
+        ];
+        var positionsBuffer = new Buffer(positions.length * 2);
+        for (var i = 0; i < positions.length; i++) {
+            positionsBuffer.writeUInt16LE(positions[i], i * 2);
+        }
+
+        var dataBuffer = Buffer.concat([indicesBuffer, positionsBuffer]);
+
+        var testGltf = {
+            "accessors": {
+                "accessor_index": {
+                    "bufferView": "index_view",
+                    "byteOffset": 0,
+                    "componentType": 5123,
+                    "count": 6,
+                    "type": "SCALAR",
+                    "extras": {
+                        "_pipeline": {}
+                    }
+                },
+                "accessor_position": {
+                    "bufferView": "position_view",
+                    "byteOffset": 0,
+                    "componentType": 5126,
+                    "count": 4,
+                    "type": "VEC3",
+                    "extras": {
+                        "_pipeline": {}
+                    }
+                }
+            },
+            "bufferViews": {
+                "position_view": {
+                    "buffer": "buffer_0",
+                    "byteOffset": 6 * 2,
+                    "byteLength": 4 * 3 * 2,

```

Comment:
Right. Sorry, that was careless of me. I also realized I got the byteLength for a lot of things wrong... I forgot that the positions use 32 bit floats instead of 16.